analyses/cms-open-data-ttbar/statistical_interference/agc_sample.py

```diff
@@ -0,0 +1,310 @@
+        histFactor.SetHistoNameLow(histoname_down)
+        histFactor.SetInputFileHigh(histofile_up)
+        histFactor.SetInputFileLow(histofile_down)
+
+        self.GetHistoFactorList().push_back(histFactor)
+
+    def AddShapeSys(
+        self,
+        name,
+        constraint_type=None,
+        histoname=None,
+        histofile=None,
+        histopath="",
+    ):  # overrided function, allowing not to provide additional input file
+
+        shapeSys = ROOT.RooStats.HistFactory.ShapeSys()
+
+        if histofile is None:
+            histofile = self.fInputFile
+
+        shapeSys.SetName(name)
+        shapeSys.SetHistoName(self.GetHistoName())
+        shapeSys.SetHistoPath(histopath)
+        shapeSys.SetInputFile(histofile)
+
+        shapeSys.SetConstraintType(constraint_type)
+
+        self.GetShapeSysList().push_back(shapeSys)
+
+    def AddNormPlusShapeHistoSys(
+        self,
+        name,
+        histoname_up=None,
+        histofile_up=None,
+        histopath_up="",  # check more here: https://github.com/scikit-hep/cabinetry/issues/26
+        histoname_down=None,
+        histofile_down=None,
+        histopath_down="",
+    ):
+        if histofile_up is None:
+            histofile_up = self.fInputFile
+            histofile_down = self.fInputFile
+            assert (
+                histofile_up != ""
+            ), "ERROR: You not specified input file for sample"
+
+        if histoname_down is None:
+            self.Symmetrize_AddNormPlusShapeHistoSys(
+                name, histoname_up, histofile_up, histopath_up
+            )
+        else:
+            self.NonSymmetrize_AddNormPlusShapeHistoSys(
+                name,
+                histoname_up,
+                histofile_up,
+                histopath_up,
+                histoname_down,
+                histofile_down,
+                histopath_down,
+            )
+
+    def Symmetrize_AddNormPlusShapeHistoSys(
+        self, name, histoname, histofile, histopath
+    ):
+        """
+        first symmertrize histogram -> calculate overallsys -> save all to sample
+        check more here:
+        https://github.com/scikit-hep/cabinetry/issues/26
+        """
+        channel_name = str(histoname.split("_")[0])
+
+        file = ROOT.TFile(histofile, "READ")
+        dir = file.GetDirectory(histopath)
+        hist_top = dir.Get(histoname)
+
+        hist_nominal_file = ROOT.TFile(self.GetInputFile(), "READ")
+        hist_nominal_name = self.GetHistoName()
+        hist_nominal_directory = hist_nominal_file.GetDirectory(
+            self.GetHistoPath()
+        )
+        hist_nominal = hist_nominal_directory.Get(hist_nominal_name)
+
+        norm_factor_up = hist_top.Integral() / hist_nominal.Integral()
+        h_new = hist_top.Clone(
+            channel_name
+            + "_"
+            + str(self.GetName())
+            + "_"
+            + name
+            + "_norm_plus_shape_up_clone"
+        )
```

Comment:
```suggestion
        h_new = hist_top.Clone(f"{channel_name}_{self.GetName()}_{name}_norm_plus_shape_up_clone")
```
I think f-strings are nicer for these large string combinations